tests/functional/test_utils.py

```diff
@@ -63,6 +63,22 @@ def test_must_return_valid_go_architecture(self):
         self.assertEqual(get_goarch(""), "amd64")
 
 
+class TestExtractTarFile(TestCase):
+    def test_extract_tarfile_unpacks_a_tar(self):
```

Comment:
question: is testing the exception (ie actually performing the attack) something that should be done here?

Reply:
good catch. I intended but forgot to add the path traversal case. Will add it.